packages/@react-aria/focus/src/focusSafely.ts

```diff
@@ -24,11 +24,12 @@ export function focusSafely(element: FocusableElement) {
   // the page before shifting focus. This avoids issues with VoiceOver on iOS
   // causing the page to scroll when moving focus if the element is transitioning
   // from off the screen.
+  const ownerDocument = getOwnerDocument(element);
   if (getInteractionModality() === 'virtual') {
-    let lastFocusedElement = document.activeElement;
+    let lastFocusedElement = ownerDocument.activeElement;
     runAfterTransition(() => {
       // If focus did not move and the element is still in the document, focus it.
-      if (document.activeElement === lastFocusedElement && document.contains(element)) {
+      if (ownerDocument.activeElement === lastFocusedElement && ownerDocument.contains(element)) {
```

Comment:
i suspect this might be another place where isConnected is needed instead
can you think of a test case for that?

Reply:
I think it's more of an "accidentally" as opposed to "supposed to"  but I'll ask if anyone in the team remembers better

Reply:
Added some tests for focusSafely, lmk if these are ok!

Reply:
From digging through the blame, it looks to be https://github.com/adobe/react-spectrum/pull/893#issue-673779973, specifically "Opening Picker/MenuTrigger with VoiceOver on iOS doesn't work because VoiceOver scrolls the page to the focused element which is off the screen, causing the tray to close." I believe. I think what you added in the tests is fine though, but this will be more of an instance where device specific testing will be needed IMO (for instance, VoiceOver's behavior with focus + scrolling may change from under us and cause a need for a different workaround).

Reply:
Should I just revert the `isConnected` change from `focusSafely` instead then? Seems like it might have the potential to cause some issues.

Reply:
quite frankly I'm not sure, I believe `isConnected` should be fine here since it is roughly equivalent to the `ownerDocument.contain` from my understanding. I'm not very familiar with it though haha, what are your concerns here?

Reply:
Mostly the "may change from under us and cause a need for a different workaround" part - don't want to break existing integrations if they upgrade their @react-aria version

Reply:
Ah, I think we should be fine, what you have currently should theoretically behave the same as the current code on main so long as users are operating within a single frame, no? With your changes it will now properly support the "portal into an iframe" case (though I think runAfterTransition will need to be updated too since it uses document.body listeners).

As for VoiceOver's behavior changing from beneath us, I think we shouldn't worry too much about it, not much we can do about to account for that IMO. If anything, we are already hacking around plenty of VoiceOver bugs and quirks so we don't really have a choice but be reactive to any updates to VoiceOver changes as time goes on. Will just need to manually test the changes in this PR is all

Reply:
Yeap, this would now support both cases, thanks for explaining!